src/log/iobuf.rs

```diff
@@ -164,30 +161,40 @@ impl IOBufs {
     pub(super) fn reserve(&self, mut raw_buf: Vec<u8>) -> Reservation {
         let buf = encapsulate(&mut *raw_buf);
         assert_eq!(buf.len() >> 32, 0);
-        assert!(buf.len() <= MAX_BUF_SZ);
-
+        assert!(buf.len() <= self.config.get_io_buf_size());
+
+        let mut printed = false;
+        macro_rules! trace_once {
+            ($($msg:expr),*) => {
+                if !printed {
+                    trace!($($msg),*);
+                    printed = true;
+                }};
+        }
```

Comment:
I wanted to use a macro here because I wanted to avoid writing `if !printed...` for each call to trace below, and macros let me pass variable numbers of arguments in, so I can retain the semantics of the `trace!` macro.